src/Compilers/VisualBasic/Portable/Symbols/MethodSignatureComparer.vb

```diff
@@ -538,8 +543,17 @@ Done:
                                     GoTo Done
                                 End If
                             End If
+                        ElseIf (comparisons And SymbolComparisonResults.TupleNamesMismatch) <> 0 AndAlso
+                            Not type1.Type.IsSameType(type2.Type, TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds) Then
```

Comment:
>TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds [](start = 66, length = 64)

The code will be more robust long term if we wouldn't make an assumption that TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds is "all - TupleNamesMismatch".